test/LoadURLTest.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2025 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>, UFR
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "engine/LoadURL.h"
+#include "util/GTestHelpers.h"
+#include "util/HttpClientTestHelpers.h"
+#include "util/IdTableHelpers.h"
+#include "util/IndexTestHelpers.h"
+#include "util/OperationTestHelpers.h"
+#include "util/RuntimeParametersTestHelpers.h"
+
```

Comment:
best put everything here (including the GTEST macros) into the anonymouse `namespace {...}`.
That gives the least linking problems and still works with test discovery etc.